gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcOperationCallableImpl.java

```diff
@@ -27,106 +27,91 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package com.google.api.gax.grpc;
+package com.google.api.gax.rpc;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 import com.google.api.core.ApiFunction;
 import com.google.api.core.ApiFuture;
-import com.google.api.core.ApiFutures;
+import com.google.api.core.InternalApi;
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.api.gax.longrunning.OperationFutureImpl;
+import com.google.api.gax.longrunning.OperationSnapshot;
 import com.google.api.gax.retrying.RetryingExecutor;
 import com.google.api.gax.retrying.RetryingFuture;
-import com.google.api.gax.rpc.ApiCallContext;
-import com.google.api.gax.rpc.ClientContext;
-import com.google.api.gax.rpc.OperationCallSettings;
-import com.google.api.gax.rpc.OperationCallable;
-import com.google.api.gax.rpc.UnaryCallable;
-import com.google.longrunning.CancelOperationRequest;
-import com.google.longrunning.GetOperationRequest;
-import com.google.longrunning.Operation;
-import com.google.longrunning.stub.OperationsStub;
-import com.google.protobuf.Any;
-import com.google.protobuf.Empty;
-import com.google.protobuf.InvalidProtocolBufferException;
-import com.google.protobuf.Message;
-import io.grpc.Status;
 
 /**
- * A GrpcOperationCallableImpl is an immutable object which is capable of initiating RPC calls to
- * long-running API methods and returning a {@link GrpcOperationFuture} to manage the polling of the
+ * An OperationCallableImpl is an immutable object which is capable of initiating RPC calls to
+ * long-running API methods and returning a {@link OperationFuture} to manage the polling of the
  * Operation and getting the response.
  *
  * <p>Package-private for internal use.
  */
-class GrpcOperationCallableImpl<RequestT, ResponseT extends Message, MetadataT extends Message>
-    extends OperationCallable<RequestT, ResponseT, MetadataT, Operation> {
-
-  private final UnaryCallable<RequestT, Operation> initialCallable;
-  private final ClientContext clientContext;
-  private final RetryingExecutor<Operation> executor;
-  private final OperationsStub operationsStub;
-  private final ApiFunction<Operation, ResponseT> responseTransformer;
-  private final ApiFunction<Operation, MetadataT> metadataTransformer;
-
-  GrpcOperationCallableImpl(
-      UnaryCallable<RequestT, Operation> initialCallable,
-      ClientContext clientContext,
-      RetryingExecutor<Operation> executor,
-      OperationsStub operationsStub,
-      OperationCallSettings<RequestT, ResponseT, MetadataT, Operation> operationCallSettings) {
+@InternalApi
+class OperationCallableImpl<RequestT, ResponseT, MetadataT>
```

Comment:
This is now in `.rpc` package, as well as the class it extends (`OperationCallable`). Do we still need `OperationCallable` and `OperationCallableImpl`? Can it just be `OperationCallable` only (two merged in one)?

Reply:
I was trying to retain the same pattern of other callable types, where there is an abstract class with the syntactic sugar (entry points that don't require the context which forward to the call which does), and then implementing classes which provide a single implementation for functionality with all the parameters.